### PR TITLE
Set RPATH for private libdirs on FreeBSD

### PR DESCRIPTION
**Pull request by ararslan**
Currently this is only done on Linux, but it's also applicable for FreeBSD.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -405,7 +405,7 @@ ifeq ($(OS), Darwin)
 		install_name_tool -rpath @executable_path/$(build_private_libdir_rel) @executable_path/$(private_libdir_rel) $$julia; \
 		install_name_tool -add_rpath @executable_path/$(build_libdir_rel) @executable_path/$(libdir_rel) $$julia; \
 	done
-else ifeq ($(OS), Linux)
+else ifneq (,$(findstring $(OS),Linux FreeBSD))
 	for julia in $(DESTDIR)$(bindir)/julia* ; do \
 		patchelf --set-rpath '$$ORIGIN/$(private_libdir_rel):$$ORIGIN/$(libdir_rel)' $$julia; \
```
Comment (tkelman) [inline Makefile]: wonder why this line isn't using the patchelf that we build?

do we build patchelf on freebsd?

Comment (ararslan) [inline Makefile]: > do we build patchelf on freebsd?

Looks like we only build it on Linux, actually.

Comment (tkelman) [inline Makefile]: ah xref https://github.com/JuliaLang/julia/pull/17901 - guess we only have our own handling of `$(PATCHELF)` wired up under deps? could probably do with some refactoring

Comment (ararslan) [inline Makefile]: I pushed a commit that builds it on FreeBSD now as well as Linux

**Comment by ararslan** [comment]
64-bit Windows failure is unrelated. Absent objections I'm inclined to merge this within the next couple days.

**Pull request merged**